accelerometer/summariseEpoch.py

```diff
@@ -423,3 +430,29 @@ def writeMovementSummaries(e, labels, summary, useRecommendedImputation):
             summary[accType + '-hourOfDay-' + str(i) + '-avg'] = hourOfDay
             summary[accType + '-hourOfWeekday-' + str(i) + '-avg'] = hourOfWeekday
             summary[accType + '-hourOfWeekend-' + str(i) + '-avg'] = hourOfWeekend
+
+
+
+def writeActivityTimeseries(e, labels, tsFile):
+    """ Write activity timeseries file
+    :param pandas.DataFrame e: Pandas dataframe of epoch data. Must contain
+        activity classification columns with missing rows imputed.
+    :param list(str) labels: Activity state labels
+    :param dict tsFile: output CSV filename
+
+    :return: None
+    :rtype: void
+    """
+    labelsImputed = [l + 'Imputed' for l in labels]
+    cols = ['accImputed'] + labelsImputed + ['METImputed']
+    cols_new = ['acc'] + labels + ['MET']
+
+    e_new = pd.DataFrame(index=e.index)
+    e_new['imputed'] = e.isna().any(1).astype('int')
+    e_new[cols_new] = e[cols]
+
+    # make output time format contain timezone
+    # e.g. 2020-06-14 19:01:15.123000+0100 [Europe/London]
+    e_new.index = e_new.index.to_series().apply(accUtils.date_strftime)
+
+    e_new.to_csv(tsFile, compression='gzip')
```

Comment:
New code that essentially replaces `accUtils.generateTimeSeries()` to write out `timeSeries.csv` with a new format: it now has a time column containing zone info.

Reply:
Again, I have a preference to keep this as a utility function. However, I there may be a better reason for moving it to summariseEpoch.py?

Reply:
Ok I can move this out of summariseEpoch. I will submit a fix soon :)